crates/assists/src/handlers/extract_module_to_file.rs

```diff
@@ -0,0 +1,94 @@
+use ide_db::source_change::FileSystemEdit;
+use vfs::FileId;
+use syntax::{
+    ast::{
+        self,
+        NameOwner
+    },
+    AstNode,
+    TextSize
+};
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+// Assist: extract_module_to_file
+//
+// This assist extract module to file.
+//
+// ```
+// mod tests {<|>
+//     #[test] fn t() {}
+// }
+// ```
+// ->
+// ```
+// mod tests;
+//
+// #[test] fn t() {}
```

Comment:
I think the way to make this work with our doctest system is to do 

```
mod foo {
    fn f() {}
}
```

->

```
mod foo;
```

Ie, make doctest just ignore (but not outright fail on) multiple files.